### PR TITLE
Task/db reconnect

### LINKED ISSUE #449: ✨ Database Reconnecting
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

The connection to the database can have network issues with a bad connector or a poor network setup.  Connections should be more resilient and allow for a graceful reconnecting.

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by SteveMaier-IRT**
[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
Fixes #449 

### Description
Added a UI to show the user that there was a network glitch that caused a database issue.   Tested on admin and guardian steps for all pages.  

### Testing
*Describe the best way to test or validate your PR.*

### CODE REVIEW

File: src/electionguard-ui/ElectionGuard.UI.Lib/Services/Database/DbService.cs [hunk 1]
```diff
@@ -46,6 +53,17 @@ public static class DbService
     private static string DbConnection = string.Empty;
     private static MongoClient client = new();
 
+    public static event EventHandler<DbEventArgs> DatabaseDisconnected;
```
Comment (john-s-morgan) [inline src/electionguard-ui/ElectionGuard.UI.Lib/Services/Database/DbService.cs:56]: Interesting pattern. I kind of like this a little better than explicit delegates

File: src/electionguard-ui/ElectionGuard.UI/Views/NetworkPopup.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace ElectionGuard.UI.Views;
+
+public partial class NetworkPopup : Popup
+{
+    public NetworkPopup()
+    {
+        InitializeComponent();
+        //var vm = Ioc.Default.GetService(typeof(ChallengedPopupViewModel));
+        //BindingContext = vm;
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard-ui/ElectionGuard.UI/Views/NetworkPopup.xaml.cs:12]: fixed

File: src/electionguard-ui/ElectionGuard.UI/ViewModels/ViewKeyCeremonyViewModel.cs [hunk 3]
```diff
@@ -49,10 +49,17 @@ public override async Task OnAppearing()
...
+        {
+            _joinPressed = await HasJoined();
+            if (!_timer.IsRunning)
+            {
+                _timer.Start();
+            }
+        }
+        catch(Exception)
+        {
+            _timer.Stop();
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard-ui/ElectionGuard.UI/ViewModels/ViewKeyCeremonyViewModel.cs:62]: This is on the onappearing and this is just for the checking of the HasJoined call.  The only thing we really need to do is to stop the timer.

File: src/electionguard-ui/ElectionGuard.UI/ViewModels/TallyProcessViewModel.cs [hunk 4]
```diff
@@ -207,18 +207,26 @@ private async Task UpdateElection(TallyRecord? newValue)
...
-        if (!(_timer?.IsRunning ?? true))
+            if (!(_timer?.IsRunning ?? true))
+            {
+                _timer.Start();
+            }
+        }
+        catch (Exception ex)
         {
-            _timer.Start();
+            ErrorMessage = ex.Message;
```
[the remark this answers is not in the document]
Comment (SteveMaier-IRT) [reply src/electionguard-ui/ElectionGuard.UI/ViewModels/TallyProcessViewModel.cs:228]: This will force the user to log in again.  the error will be clear.

**Review by john-s-morgan — APPROVED** [review]

**Pull request merged**